src/variant.rs

```diff
@@ -97,11 +101,13 @@ impl Variant {
         // See https://msdn.microsoft.com/en-us/library/cc237865.aspx for more info.
         // Rust can infer the return type of `vt.*Val()` calls,
         // but it's easier to read when the type is named explicitly.
-        let variant_value = match variant_type {
+        let variant_value = match VARENUM(variant_type) {
             VT_BSTR => {
-                let bstr_ptr: &BSTR = unsafe { &vt.Anonymous.Anonymous.Anonymous.bstrVal };
-
-                Variant::String(bstr_ptr.try_into()?)
+                let bstr_ptr = unsafe { BSTR::from_raw(vt.Anonymous.Anonymous.Anonymous.bstrVal) };
```

Comment:
Consider replacing this with

```rust
let bstr_borrow = ManuallyDrop::new(unsafe { BSTR::from_raw(vt.Anonymous.Anonymous.Anonymous.bstrVal) });
Variant::Sting(bstr_borrow.to_string())
```

This creates a non-owning `BSTR` from the get-go, so you do not have to remember and cannot forget to manually release ownership later.